nidx/nidx_paragraph/src/search_query.rs

```diff
@@ -293,20 +123,21 @@ fn filter_query(
 }
 
 pub fn suggest_query(
-    parser: &QueryParser,
-    text: &str,
     request: &ParagraphSuggestRequest,
     prefilter: &PrefilterResult,
     schema: &ParagraphSchema,
-    distance: u8,
 ) -> (Box<dyn Query>, SharedTermC, Box<dyn Query>) {
-    let mut term_collector = TermCollector::default();
-    let processed = preprocess_raw_query(text, &mut term_collector);
-    let query = parse_query(parser, &processed.regular_query);
-    let fuzzy_query = parse_query(parser, &processed.fuzzy_query);
-    let termc = SharedTermC::from(term_collector);
-    let mut fuzzies = fuzzied_queries(fuzzy_query, true, distance, termc.clone());
-    let mut originals = vec![(Occur::Must, query)];
+    let query = &request.body;
+    let parser_config = ParserConfig::with_schema(schema).last_fuzzy_query_literal_as_prefix();
+    let parsed = query_parser::parse_query(query, parser_config);
```

Comment:
It might be a bit more ergonomic to have the `parse_query` function as a method of the config:

`let parsed = ParserConfig::with_schema(schema).last_fuzzy_query_literal_as_prefix().parse(query)`

I also find `.last_fuzzy_query_literal_as_prefix()` a bit long, maybe `.last_fuzzy_term_as_prefix()`?

Anyway, these are two very nitpicky things, question of preference. Just found this couple of lines a bit long and java-ish. But it's personal preference and what's its already here is very easy to understand.

Reply:
Makes sense, we could use a `Parser` directly instead of config and the function. And like the rename, I struggled with a name that was clear enough that we were talking about the fuzzy query